pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -870,72 +889,62 @@ protected void handleConnect(CommandConnect connect) {
 
             // init authState and other var
             ChannelHandler sslHandler = ctx.channel().pipeline().get(PulsarChannelInitializer.TLS_HANDLER);
-            SSLSession sslSession = null;
             if (sslHandler != null) {
                 sslSession = ((SslHandler) sslHandler).engine().getSession();
             }
 
             authState = authenticationProvider.newAuthState(clientData, remoteAddress, sslSession);
+            doAuthentication(clientData, clientProtocolVersion, clientVersion).thenCompose(__ -> {
```

Comment:
It’s better to change `doAuthentication` to `doAuthenticationAsync`

Reply:
Done.